lib/git/aliases.sh

```diff
@@ -47,20 +47,22 @@ _alias "$git_alias" "git"
 # which I've altered slightly to be more flexible.
 # https://github.com/bronson/dotfiles/blob/731bfd951be68f395247982ba1fb745fbed2455c/.bashrc#L81
 # (only works for bash)
-__define_git_completion () {
-eval "
-_git_$1_shortcut () {
-COMP_LINE=\"git $2 \${COMP_LINE/$1 }\"
-let COMP_POINT+=$((4+${#2}-${#1}))
-COMP_WORDS=(git $2 \"\${COMP_WORDS[@]:1}\")
-let COMP_CWORD+=1
-
-local cur words cword prev
-_get_comp_words_by_ref -n =: cur words cword prev
-__git_wrap__git_main
-}
-"
-}
+if [ "$git_skip_shell_completion" != "yes" ]; then
+  __define_git_completion () {
+  eval "
+  _git_$1_shortcut () {
+  COMP_LINE=\"git $2 \${COMP_LINE/$1 }\"
+  let COMP_POINT+=$((4+${#2}-${#1}))
+  COMP_WORDS=(git $2 \"\${COMP_WORDS[@]:1}\")
+  let COMP_CWORD+=1
+
+  local cur words cword prev
+  _get_comp_words_by_ref -n =: cur words cword prev
+  __git_wrap__git_main
+  }
+  "
+  }
+fi
```

Comment:
Because this is already an eval hidden behind a function definition, I would prefer not placing this in an if condition. The calling of this function is already behind the new setting that you added.

Reply:
Good point, my inclination to have 'less stuff' but I agree this probably just increases complexity. Fix is up.